### PR TITLE
Document how to Alpakka Kafka Producer flow

### PR DESCRIPTION
**Pull request by renatocaval**
References #143

TODO: 

- [x] Java docs

### CODE REVIEW

File: docs/src/main/paradox/kafka.md [hunk 1]
```diff
@@ -118,6 +120,26 @@ Scala
 Java
 :  @@snip [KafkaDocExample.java](/examples/src/test/java/jdocs/kafka/KafkaDocExample.java) { #wordSource }
 
+### Sending to Kafka using a Producer Flow
+
+Alternatively, we can be define the same projection using @apidoc[Producer.flowWithContext](Producer$) in combination with `atLeastOnceFlow`.
```
Comment (seglo) [inline docs/src/main/paradox/kafka.md:125]:
```suggestion
Alternatively, we can define the same projection using @apidoc[Producer.flowWithContext](Producer$) in combination with `atLeastOnceFlow`.
```

File: docs/src/main/paradox/kafka.md [hunk 2]
```diff
@@ -118,6 +120,26 @@ Scala
...
+
+The `WordSource` emits `WordEnvelope`s, therefore we will build a flow that takes every single emitted `WordEnvelope` and map it using Alpakka Kafka's @apidoc[ProducerMessage.single](ProducerMessage$), we pass it through @apidoc[Producer.flowWithContext](Producer$) that will publish it to the Kafka Topic and finally we map the result to `Done`.
```
Comment (seglo) [inline docs/src/main/paradox/kafka.md:127]: I think it makes sense to elaborate the purpose of `ProducerMessage` a bit. The flexibility with producing might be a reason why you would want to use Alpakka Kafka and the flow handler to produce to Kafka.

WDYT of this:

> The `WordSource` emits `WordEnvelope`s, therefore we will build a flow that takes every single emitted `WordEnvelope` and map it into an Alpakka Kafka @apidoc[ProducerMessage]. The `ProducerMessage` factory methods can be used to produce a single message, multiple messages, or pass through a message (skip a message from being produced). The @apidoc[ProducerMessage] will through @apidoc[Producer.flowWithContext](Producer$) that will publish it to the Kafka Topic and finally we map the result to `Done`.

Comment (renatocaval) [reply docs/src/main/paradox/kafka.md:127]: yes, let's do it

Comment (renatocaval) [reply docs/src/main/paradox/kafka.md:127]: I change it to a GitHub suggestion so I can apply here directly.

Comment (renatocaval) [reply docs/src/main/paradox/kafka.md:127]: I edit it a little bit. 

It was:
> The @apidoc[ProducerMessage] will through @apidoc[Producer.flowWithContext](Producer$) that will publish it to the Kafka Topic and finally we map the result to `Done`

I change to 

> The @apidoc[ProducerMessage] will **pass** through @apidoc[Producer.flowWithContext](Producer$) that will publish it to the Kafka Topic and finally we map the result to `Done`

Note the 'pass' in "will pass through".

File: docs/src/main/paradox/kafka.md [hunk 3]
```diff
@@ -118,6 +120,26 @@ Scala
 Java
 :  @@snip [KafkaDocExample.java](/examples/src/test/java/jdocs/kafka/KafkaDocExample.java) { #wordSource }
 
+### Sending to Kafka using a Producer Flow
+
+Alternatively, we can define the same projection using @apidoc[Producer.flowWithContext](Producer$) in combination with `atLeastOnceFlow`.
+
+The `WordSource` emits `WordEnvelope`s, therefore we will build a flow that takes every single emitted `WordEnvelope` and map it into an Alpakka Kafka @apidoc[ProducerMessage]. The `ProducerMessage` factory methods can be used to produce a single message, multiple messages, or pass through a message (skip a message from being produced). The @apidoc[ProducerMessage] will pass through @apidoc[Producer.flowWithContext](Producer$) that will publish it to the Kafka Topic and finally we map the result to `Done`.
```
Comment (renatocaval) [inline docs/src/main/paradox/kafka.md:127]:
```suggestion
The `WordSource` emits `WordEnvelope`s, therefore we will build a flow that takes every single emitted `WordEnvelope` and map it into an Alpakka Kafka @apidoc[ProducerMessage$]. The `ProducerMessage` factory methods can be used to produce a single message, multiple messages, or pass through a message (skip a message from being produced). The @apidoc[ProducerMessage$] will pass through @apidoc[Producer.flowWithContext](Producer$) that will publish it to the Kafka Topic and finally we map the result to `Done`.
```

**Pull request merged**